src/binaryen-c.cpp

```diff
@@ -90,10 +90,7 @@ Literal fromBinaryenLiteral(BinaryenLiteral x) {
       return Literal(x.i64).castToF64();
     case Type::v128:
       return Literal(x.v128);
-    case Type::anyref: // there's no anyref literals
-    case Type::exnref: // there's no exnref literals
-    case Type::none:
-    case Type::unreachable:
+    default:
```

Comment:
Yeah I agree compile time errors are better than runtime ones... I'll add them back in. Maybe we can improve this by using some macro or something and factoring out the type-checking code into the macro.. I'll try that later in a follow-up patch.

Reply:
I just realized this compile time error thing does not work anymore, after we changed `Type` from an enum to a class. So there is no point listing all those types after all. I'll leave them as they are anyway because I already changed them all back. If we want to rely on the compile time error behavior, we have to come up with a new macro or something I guess.

Reply:
Oh, does the switch warning not work on the num inside the `Type` class? Seems odd, maybe I'm missing something?

Reply:
I tested this a little, and the difference is this:

```
Type::ValueType type;
switch (type) {
case i32:
  break;
}
```
Here `Type::ValueType` is an enum. This does not compile with `-Werror`, because all other types (i64, f32, f64, ...) are missing in the `switch`. But,

```
Type type;
switch (type) {
case i32:
  break;
}
```
This compiles without warnings, even though there are many types missing in the `switch`. The reason is, here `type` is not considered as an enum. It is a class object, which can be used in `switch` because it has a `constexpr operator uint32_t() const`. So it can be casted to an unsigned int, but it is not an enum, so this switch compiles without warnings. But note that the second usage is far more common in the codebase. (Do we ever explicitly use `Type::ValueType`? I don't think so.)